backends/SharedDummy/src/SharedDummyBackend.cc

```diff
@@ -130,6 +128,11 @@ namespace ChimeraTK {
 
   boost::shared_ptr<DeviceBackend> SharedDummyBackend::createInstance(
       std::string address, std::map<std::string, std::string> parameters) {
+    // create instanceId from address and parameters.
+    // note, this approach is not perfect: in case two different device URIs are created by reordering their parameters
+    // and URIs are used from same process, it will create two SharedDummyBackend instances (since URI is key in
+    // existingBackends map of factory) although it shouldn't!
```

Comment:
I think this comment should be a bit more elaborate, e.g.:

```suggestion
    // note, this approach is not perfect: in case two different device URIs are created by reordering their parameters
    // and URIs are used from same process, the BackendFactory will create two SharedDummyBackend instances (since it uses
    // the URI string as a key). The order of parameters is lost in the parameters map, so we will here assume the same
    // instanceIdHash. That will lead further down to some error, since we will try to access the same shared memory segment
    // twice in the same process.
```

I hope this summarises your findings properly?